src/server/pfs/cmds/cmds.go

```diff
@@ -1017,6 +1022,68 @@ Any pachctl command that can take a commit, can take a branch name instead.`,
 	shell.RegisterCompletionFunc(deleteBranch, shell.BranchCompletion)
 	commands = append(commands, cmdutil.CreateAliases(deleteBranch, "delete branch", branches))
 
+	FindCommits := &cobra.Command{
+		Use:   "{{alias}} <repo>@<branch-or-commit>:<path/in/pfs> [flags]",
+		Short: "search for commits with reference to <filePath> within a branch starting from <repo@commitID>",
+		Long:  "search for commits with reference to <filePath> within a branch starting from <repo@commitID>",
+		Run: cmdutil.RunFixedArgs(1, func(args []string) error {
+			file, err := cmdutil.ParseFile(project, args[0])
+			if err != nil {
+				return err
+			}
+			commit := file.Commit
+			c, err := client.NewOnUserMachine("user")
+			if err != nil {
+				return err
+			}
+			defer c.Close()
+			ctx := c.Ctx()
+			var cf context.CancelFunc
+			if timeout != time.Duration(0) {
+				ctx, cf = context.WithTimeout(c.Ctx(), timeout)
+				defer cf()
+			}
+			findCommitClient, err := c.PfsAPIClient.FindCommits(ctx, &pfs.FindCommitsRequest{
+				Start:    commit,
+				FilePath: file.Path,
+				Limit:    limit,
+			})
+			if err != nil {
+				return grpcutil.ScrubGRPC(err)
+			}
+			writer := tabwriter.NewWriter(os.Stdout, "FOUND IN COMMITS\n")
+			return grpcutil.ScrubGRPC(clientsdk.ForEachFindCommitsResponse(findCommitClient, func(x *pfs.FindCommitsResponse) error {
+				switch x.Result.(type) {
+				case *pfs.FindCommitsResponse_FoundCommit:
+					_, err := writer.Write([]byte(x.GetFoundCommit().ID + "\n"))
```

Comment:
nit: since this writer is buffered, it's probably faster to let it do the concatenation than creating a new string and passing it to the writer.

Reply:
Could you elaborate more on what you mean here by letting the writer do the concatenation?

Reply:
`writer.Write([]byte(x.GetFoundCommit().ID))`
and then 
`writer.Write("\n")`

Because it's buffered this should be efficient.  If the write made a system call than that would be a different story.